modules/installation-special-config-butane.adoc

```diff
@@ -0,0 +1,105 @@
+= Configuring machine configs with Butane
+
+Ignition config files, which control the RHCOS configuration, are used to deploy control plane and worker machines. A machine config uses an Ignition config to tell machines to create users and filesystems, set up the network, install systemd units, and more.
+
+Ignition configs are machine-readable JSON files that can be difficult to modify. You can use human-readable Butane configs to create Ignition configs, making the machine configuration process much easier.
+
+== About Butane
+
+Butane is a human-readable YAML utility that is used by {op-system} to automatically write many of the labor-intensive parts of Ignition for you. The Butane tool, based on the `butane` specification, consumes a Butane config and produces an Ignition config that can be given to a {op-system} machine when it first boots.
+
+[id="rhcos-installing-butane_{context}"]
+== Installing Butane
+
+You can install the Butane tool (`butane`) to create machine configs with {op-system} from a command-line interface. You can install `butane` on Linux, Windows, or macOS by downloading the related binary.
+
+[TIP]
+====
+Butane releases are backwards-compatible with older releases and with the Fedora CoreOS Config Transpiler (FCCT), unless otherwise noted.
+====
+
+.Procedure
+
+. Navigate to the Butane image download page at https://mirror.openshift.com/pub/openshift-v4/clients/butane/.
+. Get the `butane` binary:
+.. For the newest version of Butane, get the latest `butane` image:
++
+[source,terminal]
+----
+$ curl https://mirror.openshift.com/pub/openshift-v4/clients/butane/latest/butane
+----
++
+.. Optional: For a specific type of architecture you are installing on, such as `butane-aarch64` or `butane-ppc641e`, indicate the appropriate URL. For example:
++
+[source,terminal]
+----
+$ curl https://mirror.openshift.com/pub/openshift-v4/clients/butane/latest/butane-aarch64
+----
++
+. Unpack and unzip the archive.
+. Move the `butane` binary to a directory on your `PATH`.
++
+To check your `PATH`, open a terminal and execute the following command:
++
+[source,terminal]
+----
+$ echo $PATH
+----
+
+After you install the Butane tool, it is available by using the `butane` command:
+
+[source,terminal]
+----
+$ butane <command>
+----
+
+[id="rhcos-creating-butane_{context}"]
+== Creating an Ignition config with Butane
+
+To create machines, you can create a Butane config file that Butane consumes and transpiles into a machine-readable Ignition config file for producing machine configs.
+
+.Prerequisites
+
+* You have added your local SSH private key to the `ssh-agent`. See _Generating an SSH private key and adding it to the agent_ in the "Installing" section.
+* You have installed the `butane` utility.
+
+.Procedure
+
+. Create a Butane config file, such as `99-simple-example.bu`, as shown in the following example:
++
+[source,yaml]
+----
+variant: openshift
+version: 4.8.0
+metadata:
+  name: 99-simple-example
```

Comment:
This example won't transpile because it's missing a mandatory metadata field:

```yaml
metadata:
  name: 99-simple-example
  labels:
    machineconfiguration.openshift.io/role: worker
```